### PR TITLE
Add strict and context arguments to validation functions

### PR DESCRIPTION
**Pull request by adriangb**
TODO: docstrings

Selected Reviewer: @samuelcolvin

### CODE REVIEW

File: tests/test_validator.py [hunk 1]
```diff
@@ -92,3 +95,85 @@ def test_type_alias():
...
+    class Model(BaseModel):
+        x: int
+        y: str
+
+        model_config = ConfigDict(strict=True, title='FooModel')
+
+    analyzed = AnalyzedType(Model, config=ConfigDict(strict=False, str_max_length=20))
+
+    # strict=False gets applied to the outer Model but not to the inner typeddict validator
+    # so we're allowed to validate a dict but `x` still must be an int
```
Comment (adriangb) [inline tests/test_validator.py:166]: This is an interesting artifact of how we validate models by wrapping a typeddict schema. I would not expect this as a user. I would expect that `model_config` has the same meaning as the `config` argument to `AnalyzedType`.

File: pydantic/analyzed_type.py [hunk 2]
```diff
@@ -130,26 +132,28 @@ def __init__(self, __type: Any, *, config: ConfigDict | None = None, _parent_dep
...
             serializer = SchemaSerializer(core_schema, core_config)
 
         self.core_schema = core_schema
         self.validator = validator
         self.serializer = serializer
 
-    def validate_python(self, __object: Any) -> T:
-        return self.validator.validate_python(__object)
+    def validate_python(self, __object: Any, *, strict: bool | None = None, context: dict[str, Any] | None = None) -> T:
+        return self.validator.validate_python(__object, strict=strict, context=context)
```
Comment (samuelcolvin) [inline pydantic/analyzed_type.py:151]: need to check whether `context=None` get's passed through as a value, e.g. is different to omitting context.

**Review by samuelcolvin — COMMENTED** [review]
can you add a couple of tests for `BaseModel` with context, just to be sure it's actually working.

Otherwise LGTM.

**Comment by adriangb** [comment]
Oh yeah I'll do that

File: pydantic/analyzed_type.py (hunk 2 above)
Comment (adriangb) [reply pydantic/analyzed_type.py:151]: Should they be different? I thought it was None by default

Comment (adriangb) [reply pydantic/analyzed_type.py:151]: I confirmed that it's the same as far as the Python side is concerned

**Pull request merged**